docs/proposals/selinux.md

```diff
@@ -0,0 +1,280 @@
+3.  We will not deal with shared storage 
+
+## Current State Overview
+
+### Docker
+
+Docker uses a base SELinux context and calculates a unique MCS label per container.  The SELinux
+context of a container can be overriden with the `SecurityOpt` api that allows setting the different
+parts of the SELinux context individually.
+
+Docker has functionality to relabel bind-mounts with a usable SElinux and supports two different
+use-cases:
+
+1.  The `:Z` bind-mount flag, which tells Docker to relabel a bind-mount with the container's
+    SELinux context
+2.  The `:z` bind-mount flag, which tells Docker to relabel a bind-mount with the container's
+    SElinux context, but remove the MCS labels, making the volume shareable beween containers
+
+We should avoid using the `:z` flag, because it relaxes the SELinux context so that any container
+(from an SELinux standpoint) can use the volume.
+
+### Rocket
+
+Rocket currently reads the base SELinux context to use from `/etc/selinux/*/contexts/lxc_contexts`
+and allocates a unique MCS label per pod.
+
+### Kubernetes
+
+
+There is a [proposed change](https://github.com/GoogleCloudPlatform/kubernetes/pull/9844) to the
+EmptyDir plugin that adds SELinux relabeling capabilities to that plugin, which is also carried as a
+patch in [OpenShift](https://github.com/openshift/origin).  It is preferable to solve the problem
+in general of handling SELinux in kubernetes to merging this PR.
+
+A new `PodSecurityContext` type has been added that carries information about security attributes
+that apply to the entire pod and that apply to all containers in a pod.  See:
+
+1.  [Skeletal implementation](https://github.com/kubernetes/kubernetes/pull/13939)
+1.  [Proposal for inlining container security fields](https://github.com/kubernetes/kubernetes/pull/12823)
+
+## Use Cases
+
+1.  As a cluster operator, I want to support securing pods from one another using SELinux when
+    SELinux integration is enabled in the cluster
+2.  As a user, I want volumes sharing to work correctly amongst containers in pods 
+
+#### SELinux context: pod- or container- level?
+
+Currently, SELinux context is specifiable only at the container level.  This is an inconvenient
+factoring for sharing volumes and other SELinux-secured resources between containers because there
+is no way in SELinux to share resources between processes with different MCS labels except to
+remove MCS labels from the shared resource.  This is a big security risk: _any container_ in the
+system can work with a resource which has the same SELinux context as it and no MCS labels.  Since
+we are also not interested in isolating containers in a pod from one another, the SELinux context
+should be shared by all containers in a pod to facilitate isolation from the containers in other
+pods and sharing resources amongst all the containers of a pod.
+
+#### Volumes
+
+Kubernetes volumes can be divided into two broad categories:
+
+1.  Unshared storage:
```

Comment:
To expand a little bit more, if you're using a block device-based volume that supports `chcon` in one of the `Many` PV modes, I think you have problems similar to 'shared' FSs like NFS, etc:

1.  Which context is the right one to use?
2.  How should manage the context?

I'm inclined to say the right point at which to solve the above questions is when such a volume is provisioned rather than when it is mounted into a pod.